sw/device/silicon_creator/lib/epmp_unittest.cc

```diff
@@ -0,0 +1,346 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/epmp.h"
+
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/base/testing/mock_csr.h"
+
+namespace epmp_unittest {
+using ::mock_csr::CsrTest;
+using ::testing::Test;
+
+/**
+ * PMP control registers.
+ */
+struct PmpCsrs {
+  std::vector<uint8_t> pmpcfg = std::vector<uint8_t>(kEpmpNumRegions, 0);
```

Comment:
You can write `T x(args);` instead of `T x = T(args);` Yay C++

Reply:
I can't actually do this here because `pmpcfg` is then parsed as a method declaration. Yay C++ indeed :)

Instead I've swapped `std::vector` out for `std::array` which is better anyway.